### PR TITLE
[jit] Add an intrinsic for RuntimeHelpers.SufficientExecutionStack (). Only supported for amd64 linux/osx for now.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by jaykrell** [comment]
I understand there's a slight inaccuracy/perf otherwise, but must it be intrinsic?

**Comment by vargaz** [comment]
The icall version is actually kinda slow since it has to go through the wrapper etc.

File: mono/mini/intrinsics.c [hunk 1]
```diff
@@ -468,8 +468,32 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+			stack_bottom_reg = alloc_preg (cfg);
+			EMIT_NEW_BIALU (cfg, ins, OP_PSUB, stack_bottom_reg, stack_end_reg, stack_size_reg);
+			sp_reg = alloc_preg (cfg);
+			MONO_INST_NEW (cfg, ins, OP_GET_SP);
+			ins->dreg = sp_reg;
+			MONO_ADD_INS (cfg->cbb, ins);
+			diff_reg = alloc_preg (cfg);
+			EMIT_NEW_BIALU (cfg, ins, OP_PSUB, diff_reg, sp_reg, stack_bottom_reg);
+			EMIT_NEW_UNALU (cfg, ins, OP_PCONV_TO_I4, diff_reg, diff_reg);
+			MONO_EMIT_NEW_BIALU_IMM (cfg, OP_ICOMPARE_IMM, -1, diff_reg, 131072);
```
Comment (lewurm) [inline mono/mini/intrinsics.c]: what magic number is `131072`?

Comment (vargaz) [reply mono/mini/intrinsics.c]: 128kb.

Comment (luhenry) [reply mono/mini/intrinsics.c]: Could you replace it by `128 * 1024`, so it documents itself?

**Comment by kumpera** [comment]
The issue with the wrapper is that it didn't work on Android/linux.

File: mono/mini/intrinsics.c (hunk 1 above)
Comment (kumpera) [reply mono/mini/intrinsics.c]: Why 128Kb?

File: mono/mini/cpu-amd64.md [hunk 2]
```diff
@@ -824,5 +824,7 @@ gc_param_slot_liveness_def: len:0
 
 generic_class_init: src1:A len:32 clob:c
 get_last_error: dest:i len:32
+get_sp: dest:i len:6
+set_sp: src1:i len:6
```
Comment (lewurm) [inline mono/mini/cpu-amd64.md]: where is `set_sp` used?

**Comment by akoeplinger** [comment]
The sufficient-stack-intrins.exe test fails on Windows x64: https://jenkins.mono-project.com/job/x/15514/

**Comment by steveisok** [comment]
@vargaz what's the status of this PR?

**Comment by vargaz** [comment]
Could be closed i guess.

**Pull request closed without merging**